crates/builtin/build.rs

```diff
@@ -98,6 +110,11 @@ impl CommandExt for Command {
     }
 }
 
+#[derive(Debug, Deserialize)]
+struct DumpLanguage {
+    builtins: DumpBuiltins,
```

Comment:
The new CLI splits all builtins into `builtins` for functions and `constants` for values, and they have different structures. We should parse both of them with their kind information to replace the hard-coded names [here](https://github.com/oxalica/nil/blob/f6bd5fea84bb5b0b3685b5dd9c21c83fe6ab9b64/crates/builtin/build.rs#L66-L69).

I think we can move the `Command` call to `dump*` and the parsing code into a individual function. So the huge `map_or_else` above can also be refactored into an early return.

Reply:
I'm honestly not sure what you mean.
I've compared the old `__dump-builtins` to `__dump-language`'s `builtins` property, and there's two differences - the new CLI has 4 more builtins - `findFile`, `fromTOML`, `getContext`, `hasContext`, and every value has an `experimental-feature` key under it.

Reply:
Constants like `true` and `storeDir` are documented under the new `constants` property, not `builtins` property, while they are not documented before. They should also be included.

Reply:
Oh, I see. I'm not that proficient in Rust, so I might need some help with that...